include/swift/AST/DiagnosticConsumer.h

```diff
@@ -94,11 +94,12 @@ class DiagnosticConsumer {
   /// \param FormatArgs The diagnostic format string arguments.
   ///
   /// \param Info Extra information associated with the diagnostic.
-  virtual void handleDiagnostic(SourceManager &SM, SourceLoc Loc,
-                                DiagnosticKind Kind,
-                                StringRef FormatString,
-                                ArrayRef<DiagnosticArgument> FormatArgs,
-                                const DiagnosticInfo &Info) = 0;
+  virtual void
+  handleDiagnostic(SourceManager &SM, SourceLoc Loc, DiagnosticKind Kind,
+                   StringRef FormatString,
+                   ArrayRef<DiagnosticArgument> FormatArgs,
+                   const DiagnosticInfo &Info,
+                   SourceLoc bufferIndirectlyCausingDiagnostic) = 0;
```

Comment:
Can you document the parameter too? It's pretty subtle for someone writing their own DiagnosticConsumer. (It may even be worth mentioning that it's safe to ignore if your consumer doesn't do anything special with different files.)

Reply:
You could mention `setBufferIndirectlyCausingDiagnosticToInput` too.

Reply:
Good idea! Done, but a little verbosely.